platform/darwin/src/http_file_source.mm

```diff
@@ -89,7 +89,17 @@ void cancel() {
             sessionConfig.requestCachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
             sessionConfig.URLCache = nil;
 
-            session = [NSURLSession sessionWithConfiguration:sessionConfig];
+            // app might provide a NSURLSessionDelegate for authentication and such
+            id<NSURLSessionDelegate> sessionDelegate = nil;
+            NSDictionary *info = [[NSBundle mainBundle] infoDictionary];
+            NSString *delegateClassName = [info valueForKey:@"MGLNSURLSessionDelegateClassName"];
+            if (delegateClassName) {
+                sessionDelegate = [[NSClassFromString(delegateClassName) alloc] init];
```

Comment:
This is great! Would you mind documenting `MGLNSURLSessionDelegateClassName` in the following two documents?

https://github.com/mapbox/mapbox-gl-native/blob/master/platform/ios/docs/guides/Info.plist%20Keys.md
https://github.com/mapbox/mapbox-gl-native/blob/master/platform/macos/docs/guides/Info.plist%20Keys.md

Some points that could be covered, since I think this is the first time we’ve introduced an API that represents a class name as a string:

* If the class is implemented in Swift, it needs to bridge to Objective-C.
* The key needs to be set to a runtime class name. (A Swift application would either need to give the class an Objective-C name or set the key to a mangled name.)
* The class should conform to the NSURLSessionDelegate protocol. (Since the protocol’s methods are all optional, we can’t check for conformance at runtime. If the class doesn’t conform to the protocol, there wouldn’t be any adverse effects, but that would probably be a mistake on the part of the developer.)
* See also `MGLOfflineStorageDelegate`. (Like the new Info.plist key, that protocol is used for any request initiated by mbgl but presumably not for Mapbox Telemetry requests, which are handled by a different dependency.)

Reply:
@halset are you planning on updating this PR?

Reply:
@julianrex I still want this to be merged, but the documentation work suggested by @1ec5 has not bubbled up to the top of my priority list yet. Sorry.

Reply:
No worries! Was just checking in 🙂

Reply:
Perhaps we could add a method `mgl_applicationInfoDictionary` to [NSBundle+MGLAdditions.m](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/src/NSBundle%2BMGLAdditions.m) and use that here instead.

Reply:
@datwelk good idea. I've added your comment as a new issue: https://github.com/mapbox/mapbox-gl-native/issues/13908 - so that we can keep that as a separate PR.